### PR TITLE
LPS-169851 Change all fields request to get by external reference code instead of ID

### PR DESCRIPTION
**Pull request by danvb112**
Related Issue: https://issues.liferay.com/browse/LPS-169851

### CODE REVIEW

**Comment by danvb112** [comment]
This PR is blocked because it is a dependency of https://github.com/liferay-objects/liferay-portal/pull/1436

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:objects

**Comment by danvb112** [comment]
ci:test:object

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx [hunk 1]
```diff
@@ -588,11 +603,16 @@ export default function ActionBuilder({
...
+									error={
+										errors.objectDefinitionExternalReferenceCode
+									}
 									onChange={handleSelectObject}
 									options={objectsOptions}
 									value={
-										values.parameters?.objectDefinitionId
+										values.parameters
+											?.objectDefinitionExternalReferenceCode
+											? `${values.parameters?.objectDefinitionExternalReferenceCode},${values.parameters?.objectDefinitionId}`
```
Comment (naraandrade) [inline modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/tabs/ActionBuilder.tsx:614]: @danvb112 I think you can remove the optional chaining on `${values.parameters?.objectDefinitionExternalReferenceCode}...`

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Comment by danvb112** [comment]
ci:test:sf

File: modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/api.ts [hunk 2]
```diff
@@ -186,12 +186,28 @@ export async function getObjectFields(objectDefinitionId: number) {
 	);
 }
 
+export async function getObjectFieldsByExternalReferenceCode(
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/apps/object/object-js-components-web/src/main/resources/META-INF/resources/utils/api.ts:207]: Make total sense @naraandrade, Thanks!

File: modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectRelationship/SelectRelationship.tsx [hunk 3]
```diff
@@ -42,38 +47,66 @@ export default function SelectRelationship({
...
 		else {
 			setFields([]);
 		}
-	}, [objectDefinitionId]);
+	}, [objectDefinitionExternalReferenceCode]);
 
 	return (
-		<Select
+		<AutoComplete<LabelNameObject>
+			emptyStateMessage={Liferay.Language.get('no-parameter-were-found')}
```
[the remark this answers is not in the document]
Comment (danvb112) [reply modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectRelationship/SelectRelationship.tsx:86]: Thanks for noticing this, @naraandrade! :smile:

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

File: modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties [hunk 4]
```diff
@@ -10704,6 +10704,7 @@ no-outstanding-invites=No outstanding invites.
 no-page-template-could-be-imported=No page template could be imported.
 no-pages=No Pages
 no-pages-were-found-that-matched-the-keywords-x=No pages were found that matched the keywords: {0}.
+no-parameter-were-found=No parameter were found.
```
Comment (naraandrade) [inline modules/apps/portal-language/portal-language-lang/src/main/resources/content/Language.properties:10707]: @danvb112 shouldn't it be "No parameters were found"? If we are referring to just one parameter, then it should be "No parameter was found".

**Review by naraandrade — CHANGES_REQUESTED** [review]

**Comment by danvb112** [comment]
ci:test:sf

**Comment by danvb112** [comment]
ci:test:relevant

**Comment by danvb112** [comment]
ci:test:relevant

**Review by naraandrade — APPROVED** [review]

**Comment by danvb112** [comment]
Sent to: https://github.com/brianchandotcom/liferay-portal/pull/127613

**Pull request closed without merging**